### PR TITLE
[SPARK-29948][SQL] make the default alias consistent between date, timestamp and interval

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Update `Literal.sql` to make date, timestamp and interval consistent. They should all use the `TYPE 'value'` format.

### Why are the changes needed?

Make the default alias consistent. For example, without this patch we will see
```
scala> sql("select interval '1 day', date '2000-10-10'").show
+------+-----------------+
|1 days|DATE '2000-10-10'|
+------+-----------------+
|1 days|       2000-10-10|
+------+-----------------+
```

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
@maropu @MaxGekk @yaooqinn

File: sql/core/src/test/resources/sql-tests/results/interval-display-iso_8601.sql.out [hunk 1]
```diff
@@ -16,6 +16,6 @@ SELECT
   interval '1 years 2 months -3 days 4 hours 5 minutes 6.789 seconds',
   - interval '1 years 2 months -3 days 4 hours 5 minutes 6.789 seconds'
 -- !query 0 schema
-struct<CAST(NULL AS INTERVAL):interval,0 seconds:interval,1 years:interval,1 months:interval,1 years 2 months:interval,1 days -1 hours:interval,-1 days -1 hours:interval,-1 days 1 hours:interval,-1 days 1 hours:interval,1 years 2 months -3 days 4 hours 5 minutes 6.789 seconds:interval,-1 years -2 months 3 days -4 hours -5 minutes -6.789 seconds:interval>
+struct<CAST(NULL AS INTERVAL):interval,INTERVAL '0 seconds':interval,INTERVAL '1 years':interval,INTERVAL '1 months':interval,INTERVAL '1 years 2 months':interval,INTERVAL '1 days -1 hours':interval,INTERVAL '-1 days -1 hours':interval,INTERVAL '-1 days 1 hours':interval,INTERVAL '-1 days 1 hours':interval,INTERVAL '1 years 2 months -3 days 4 hours 5 minutes 6.789 seconds':interval,INTERVAL '-1 years -2 months 3 days -4 hours -5 minutes -6.789 seconds':interval>
```
Comment (yaooqinn) [inline sql/core/src/test/resources/sql-tests/results/interval-display-iso_8601.sql.out]: in this case, we may need fix the `--SET ` issue first

Comment (cloud-fan) [reply sql/core/src/test/resources/sql-tests/results/interval-display-iso_8601.sql.out]: you are right. I'll regenerate these after your fix is merged.

**Review by MaxGekk — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala [hunk 2]
```diff
@@ -407,9 +407,10 @@ case class Literal (value: Any, dataType: DataType) extends LeafExpression {
     case (v: Long, TimestampType) =>
       val formatter = TimestampFormatter.getFractionFormatter(
         DateTimeUtils.getZoneId(SQLConf.get.sessionLocalTimeZone))
-      s"TIMESTAMP('${formatter.format(v)}')"
+      s"TIMESTAMP '${formatter.format(v)}'"
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala]: There is a test case corresponding to this.
```
"TIMESTAMP[ '2019-03-21 01:02:03.456']" did not equal "TIMESTAMP[('2019-03-21 01:02:03.456')]"
```

**Review by HyukjinKwon — APPROVED** [review]

**Pull request closed without merging**